core/src/main/scala/pickling/pickler/JavaBigDecimal.scala

```diff
@@ -0,0 +1,34 @@
+package scala.pickling
+package pickler
+
+import java.math.BigDecimal
+
+/** Contains picklers which serialize java.math.BigDecimal.
+  * Note; This currently serialzies as a string.
+  */
+trait JavaBigDecimalPicklers { self: PrimitivePicklers =>
```

Comment:
So this trait is pulling in the `stringPickler` as a value, which means it *need* to be instantiated BEFORE this trait.   Which means you need the `is-a` relationship for that to be enforced.

Additionally, the ONLY way to override the default string pickler, for this trait, is via early-initializer.   You can't just use `override val`.


SO, again I ask:   What's the benefit and does it outweigh the cost of users running into runtime NPEs because of trait ordering issues?

Cake is hard, let's not make it harder by using fancy self-types when they're not providing real value.